File: algorithm-leetcode-jvm/src/test/java/org/muxun/algorithm/leetcode/java/binarytree/LeetCode1026Test.java
package org.muxun.algorithm.leetcode.java.binarytree;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

/**
 * @author 慕勋
 * @created 2024-04-05
 */
@DisplayName("节点与其祖先之间的最大差值测试用例")
public class LeetCode1026Test {

    @Test
    public void testOne() {
        TreeNode root = new TreeNode(8);
        root.left = new TreeNode(3);
        root.right = new TreeNode(10);
        root.left.left = new TreeNode(1);
        root.left.right = new TreeNode(6);
        root.left.right.left = new TreeNode(4);
        root.left.right.right = new TreeNode(7);
        root.right.right = new TreeNode(14);
        root.right.right.left = new TreeNode(13);
        int expected = 7;
        int res = new LeetCode1026().maxAncestorDiff(root);
        Assertions.assertEquals(expected, res);
    }

    @Test
    public void testTwo() {
        TreeNode root = new TreeNode(1);
        root.right = new TreeNode(2);
        root.right.right = new TreeNode(0);
        root.right.right.left = new TreeNode(3);
        int expected = 3;
        int res = new LeetCode1026().maxAncestorDiff(root);
        Assertions.assertEquals(expected, res);
    }
}

File: algorithm-leetcode-python/tests/binary_tree/LeetCode1026Test.py
import unittest

from src.binary_tree.LeetCode1026 import LeetCode1026
from src.binary_tree.TreeNode import TreeNode


class LeetCode1026Test(unittest.TestCase):
    def test_one(self):
        root = TreeNode(8)
        root.left = TreeNode(3)
        root.right = TreeNode(10)
        root.left.left = TreeNode(1)
        root.left.right = TreeNode(6)
        root.right.right = TreeNode(14)
        root.left.right = TreeNode(4)
        root.left.right.right = TreeNode(7)
        root.right.right.left = TreeNode(13)
        expected = 7
        res = LeetCode1026().maxAncestorDiff(root)
        self.assertEquals(expected, res)

    def test_two(self):
        root = TreeNode(1)
        root.right = TreeNode(2)
        root.right.right = TreeNode(0)
        root.right.right.left = TreeNode(3)
        expected = 3
        res = LeetCode1026().maxAncestorDiff(root)
        self.assertEquals(expected, res)
